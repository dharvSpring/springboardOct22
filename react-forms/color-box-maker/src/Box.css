.Box {
    margin: 1rem;
}