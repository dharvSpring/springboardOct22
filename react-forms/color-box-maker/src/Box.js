import React from "react";
import './Box.css'

function Box({id, handleRemove, width=100, height=100, backgroundColor="cornflowerblue"}) {
    const boxStyle = {
        backgroundColor: backgroundColor,
        width: `${width}px`,
        height: `${height}px`,
    }
    const removeBox = () => {
        handleRemove(id);
    }

    return (
        <div className="Box">
            <div style={boxStyle} />
            <button onClick={removeBox}>Remove Me!</button>
        </div>
    )
}

export default Box;