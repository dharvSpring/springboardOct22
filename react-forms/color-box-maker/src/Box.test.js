import React from "react";
import { render } from "@testing-library/react";

import Box from "./Box";

it("smoketest", function() {
    render(<Box />);
});

it("snapshot", function() {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});
