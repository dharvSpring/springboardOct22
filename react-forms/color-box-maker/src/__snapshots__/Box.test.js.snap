// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="background-color: cornflowerblue; width: 100px; height: 100px;"
    />
    <button>
      Remove Me!
    </button>
  </div>
</DocumentFragment>
`;
